File: BinarySearch/875_KokoEatingBananas/main.py
from re import M



piles = [3,6,7,11]
h = 8

def minEatingSpeed(piles, h):
        """
        :type piles: List[int]
        :type h: int
        :rtype: int
        Input: piles = [3,6,7,11], h = 8
        Output: 4
        """
        l,r= 1, max(piles)
        while l < r:
            m = (l + r) // 2
            if sum((p + m - 1) / m for p in piles) > h:
                l = m + 1
            else:
                r = m
        return l

print(minEatingSpeed(piles, h))
        

File: BinarySearch/875_KokoEatingBananas/main.cpp
int minEatingSpeed(vector<int>& piles, int h) {
        int l = 1;
        int r = 1000000000;
        // for(auto pile:piles){
        //     if (pile > r) r = pile;
        // }
        while(l<r){
            int m = l+(r-l)/2;
            int count = 0;
            for(auto pile: piles){
                count += (pile+m-1)/m;
            }
            if(count>h) l=m+1;
            else r = m;
        }
        return l;
    }